fdb-extensions/src/main/java/com/apple/foundationdb/async/rtree/NodeKind.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * NodeKind.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2023 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.async.rtree;
+
+import com.google.common.base.Verify;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Enum to capture the kind of node.
+ */
+public enum NodeKind {
+    LEAF((byte)0x00),
+    INTERMEDIATE((byte)0x01);
+
+    private final byte serialized;
```

Comment:
since we do variable length encoding it probably can be a `long`. Note that this field is the place where I was going to encode the version of the index.

Reply:
So the code path that does the `(byte)t.getLong(...)` has a switch in it that requires this to be cast into an `int` or  `byte` anyway. So changing this to become a true long would require me to write `if-then-elseif` branching which I think is not worth it.